Sources/FeaturevisorSDK/Instance.swift

```diff
@@ -314,11 +314,46 @@ public class FeaturevisorInstance {
     }
 
     func startRefreshing() {
-        // TODO: write implementation
+
+        guard let datafileUrl else {
+            logger.error("cannot start refreshing since `datafileUrl` is not provided")
+            return
+        }
+
+        guard timer == nil else {
+            logger.warn("refreshing has already started")
+            return
+        }
+
+        guard let refreshInterval else {
+            logger.warn("no `refreshInterval` option provided")
+            return
+        }
+
+        DispatchQueue.global().async { [weak self] in
+            self?.timer = Timer.scheduledTimer(withTimeInterval: TimeInterval(refreshInterval), repeats: true) { _ in
+                self?.refresh()
+            }
+
+            RunLoop.current.add((self?.timer)!, forMode: .common)
```

Comment:
Hmm maybe you can have a `let timer` assign it to self and here with our `!` - I mean you can easily avoid that